PowerPointLabs/PowerPointLabs/TimerLab/TimerLabPaneWPF.xaml.cs

```diff
@@ -439,7 +439,23 @@ private void WidthTextBox_Loaded(object sender, RoutedEventArgs e)
 
         private void WidthTextBox_PreviewTextInput(object sender, System.Windows.Input.TextCompositionEventArgs e)
         {
-            e.Handled = IsNumbersOnly(e.Text);
+            e.Handled = !IsNumbersOnly(e.Text);
+        }
+
+        private void WidthTextBox_TextBoxPasting(object sender, DataObjectPastingEventArgs e)
+        {
+            if (e.DataObject.GetDataPresent(typeof(String)))
+            {
+                String text = (String)e.DataObject.GetData(typeof(String));
+                if (!IsNumbersOnly(text))
+                {
+                    e.CancelCommand();
```

Comment:
This action and the one on line 442 do practically the same thing. We should have some consistency and use either one method or the other, not both.